src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSMultiBatchSink.java

```diff
@@ -92,12 +99,31 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
 
     PluginProperties formatProperties = formatPropertiesBuilder.build();
 
-    OutputFormatProvider outputFormatProvider =
-      pipelineConfigurer.usePlugin(ValidatingOutputFormat.PLUGIN_TYPE, format, FORMAT_PLUGIN_ID, formatProperties);
-    if (outputFormatProvider == null) {
-      collector.addFailure(
-        String.format("Could not find the '%s' output format plugin.", format), null)
-        .withPluginNotFound(FORMAT_PLUGIN_ID, format, ValidatingOutputFormat.PLUGIN_TYPE);
+    if (!this.config.containsMacro("format")) {
+      String format = config.getFormatName();
+      OutputFormatProvider outputFormatProvider =
+              pipelineConfigurer.usePlugin(ValidatingOutputFormat.PLUGIN_TYPE, format, FORMAT_PLUGIN_ID,
+                      formatProperties);
+      if (outputFormatProvider == null) {
+        collector.addFailure(
+                String.format("Could not find the '%s' output format plugin.", format), null)
+                .withPluginNotFound(FORMAT_PLUGIN_ID, format, ValidatingOutputFormat.PLUGIN_TYPE);
+      }
+    } else {
+      //deploying all format plugins if its macro, so that required format plugin is available when macro is resolved
+      FileFormat[] var3 = FileFormat.values();
+
+      for (FileFormat f:var3) {
```

Comment:
nit : `for (FileFormat f: FileFormat.values()) {`

Reply:
done